client/python/cryoet_data_portal/tests/test_get_client.py

```diff
@@ -84,3 +86,20 @@ def test_item_relationship_with_missing_id(client: Client):
     assert anno
     assert anno.deposition_id is None
     assert anno.deposition is None
+
+
+# Make sure we can fetch data via a thread pool without errors
```

Comment:
Just to confirm that the scope of this test is to check that the client can be used from a different thread than it was created in? And not to test concurrent requests on multiple threads, which this test may do, but without guarantees.

If that's the case, then I think you could just do something as simple as

```python
    def _make_request(client: Client):
        return Annotation.find(client)

    with ThreadPoolExecutor(max_workers=1) as tpe:
        future = tpe.submit(_make_request, client)
        assert(len(future.result()) == 6)
```

Or if you also want a pre-condition that the client has alread made one request, then you could do so before submitting one on another thread.

Most of the test feels like its testing `ThreadPoolExecutor` rather the intended behavior.

Reply:
The code you posted works fine even with the old client -- The client could always be used from a different thread than it was created in. The problem was concurrent access to the transport, so this test tries to **force** that concurrent access. In practice, it seems to accomplish that since I've never seen this succeed with the old client code. I realize that the test in the PR doesn't *guarantee* that we forced the problem state to happen though -- do you know if there's a way to accomplish that?

Reply:
Thanks for explaining.

One way to guarantee it is by implementing our own `Transport` that extends the functionality of `RequestsHTTPTransport` and that also uses concurrency primitives, so that we can control the execution of `connect` and `close` in the test to guarantee concurrent requests. Then providing some way to use that at test time (e.g. by passing through an optional initializer argument), or by monkey-patching it in the test.

If were writing lots of tests that needed these guarantees, it would be worth it. But seems like overkill for this one test, especially when it won't flake as a false positive (only as a false negative with some small probability).